frame/stake-tracker/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+pub(crate) mod mock;
+#[cfg(test)]
+mod tests;
+
+use frame_election_provider_support::{SortedListProvider, VoteWeight};
+use frame_support::traits::{Currency, CurrencyToVote};
+pub use pallet::*;
+
+use sp_staking::{OnStakingUpdate, Stake, StakingInterface};
+
+use sp_std::vec::Vec;
+
+/// The balance type of this pallet.
+pub type BalanceOf<T> = <<T as Config>::Staking as StakingInterface>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use crate::*;
+	use frame_election_provider_support::{SortedListProvider, VoteWeight};
+	use frame_support::pallet_prelude::*;
+
+	use sp_staking::StakingInterface;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// This has to come from Staking::Currency
+		type Currency: Currency<Self::AccountId, Balance = BalanceOf<Self>>;
+
+		type Staking: StakingInterface<AccountId = Self::AccountId>;
+
+		type VoterList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	/// The total balance that can be slashed from a stash account as of right now.
+	pub fn slashable_balance_of(who: &T::AccountId) -> BalanceOf<T> {
```

Comment:
slashing is a staking concept -- we should rename this to something like `active_stake_of`